### DISCUSSION TITLE
[FEATURE]建议可以禁止容器访问外网

### DISCUSSION THREAD

**Discussion by dididaguai** [1Panel-dev/1Panel#6693] [Ideas]
### 1Panel 版本

v1.10.18-lts

### 请描述您的需求或者改进建议

可以在应用创建时选择是否允许应用访问外网

### 请描述你建议的实现方案

添加是否禁止访问外网的选项

**Comment by pooneyy** [comment]
![image](https://github.com/user-attachments/assets/91684d44-9a14-4f26-b4a6-ee5f0631adef)
这样子设置

**Comment by wanghe-fit2cloud** [comment]
也可以使用 `docker network create --internal` 命令创建一个内部网络，该网络默认不允许访问外部互联网。
```shell
docker network create --internal isolated_net
```
这会创建一个隔离网络，容器只能与该网络内的其他容器通信，无法访问外网。

> 应用商店中的应用使用了默认的 `1panel-network`（bridge 网络），你可以在部署应用时，通过高级选项手动设置自定义网卡。

**Comment by dididaguai** [comment]
就是说想实现，外部可以访问容器，但是容器不能主动访问外网那种

**Comment by dididaguai** [comment]
> 也可以使用 `docker network create --internal` 命令创建一个内部网络，该网络默认不允许访问外部互联网。
> 
> ```shell
> docker network create --internal isolated_net
> ```
> 
> 这会创建一个隔离网络，容器只能与该网络内的其他容器通信，无法访问外网。
> 
> > 应用商店中的应用使用了默认的 `1panel-network`（bridge 网络），你可以在部署应用时，通过高级选项手动设置自定义网卡。

这种配置后，外面无法访问容器了。

**Comment by wanghe-fit2cloud** [comment]
> > 也可以使用 `docker network create --internal` 命令创建一个内部网络，该网络默认不允许访问外部互联网。
> > ```shell
> > docker network create --internal isolated_net
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 这会创建一个隔离网络，容器只能与该网络内的其他容器通信，无法访问外网。
> > > 应用商店中的应用使用了默认的 `1panel-network`（bridge 网络），你可以在部署应用时，通过高级选项手动设置自定义网卡。
> 
> 这种配置后，外面无法访问容器了。

宿主机与容器之间的通信不受 --internal 标志的限制，因为宿主机有对容器的直接访问权限。

**Comment by dididaguai** [comment]
> ```shell
> docker network create --internal isolated_net
> ```

![image](https://github.com/user-attachments/assets/9a2d783c-eba7-4610-b859-afb083d7c109)
![image](https://github.com/user-attachments/assets/ae6fad8f-0e96-4446-ac36-b281c392a473)
![image](https://github.com/user-attachments/assets/22781c8f-a37f-4441-93f0-f8c6d11d1abf)
![image](https://github.com/user-attachments/assets/1b380977-2995-4872-8c66-394997b77ecb)
![image](https://github.com/user-attachments/assets/3f0c83f0-751e-4893-8724-6a122b2dd5e1)
配置后，无法通过映射端口访问容器了

**Comment by wanghe-fit2cloud** [comment]
> 配置后，无法通过映射端口访问容器了

- `--internal` 网络无法对外暴露端口，适用于完全隔离的内部通信。
- 如需暴露端口，应该使用其他类型的 Docker 网络配置（如 bridge 网络），并配置适当的防火墙规则来限制访问。

**Comment by Muska-Ami** [comment]
https://blog.1l1.icu/2024/10/10/docker-jin-zhi-jiang-rong-qi-duan-kou-ying-she-dao-gong-wang/
可能对你有帮助

**Reply by Muska-Ami** [reply]
草，禁止容器访问外网，看错了（
